src/NuGet.Core/NuGet.Protocol/README.md

```diff
@@ -1 +1,190 @@
+
+```c#
+SourceRepository repository = Repository.Factory.GetCoreV3("https://api.nuget.org/v3/index.json");
+PackageSearchResource resource = await repository.GetResourceAsync<PackageSearchResource>();
+SearchFilter searchFilter = new SearchFilter(includePrerelease: true);
+
+IEnumerable<IPackageSearchMetadata> results = await resource.SearchAsync(
+    "json",
+    searchFilter,
+    skip: 0,
+    take: 20,
+    NullLogger.Instance,
+    CancellationToken.None);
+
+foreach (IPackageSearchMetadata result in results)
+{
+    Console.WriteLine($"Found package {result.Identity.Id} {result.Identity.Version}");
+}
+```
+
+### Push a package
+
+Push a package using the [NuGet V3 Push and Delete API](https://learn.microsoft.com/nuget/api/package-publish-resource):
+
+```c#
+SourceRepository repository = Repository.Factory.GetCoreV3("https://api.nuget.org/v3/index.json");
+PackageUpdateResource resource = await repository.GetResourceAsync<PackageUpdateResource>();
+
+string apiKey = "my-api-key";
+
+await resource.Push(
+    "MyPackage.nupkg",
+    symbolSource: null,
+    timeoutInSecond: 5 * 60,
+    disableBuffering: false,
+    getApiKey: packageSource => apiKey,
+    getSymbolApiKey: packageSource => null,
+    noServiceEndpoint: false,
+    skipDuplicate: false,
+    symbolPackageUpdateResource: null,
+    NullLogger.Instance);
+```
+
+### Work with authenticated feeds
+
+Use [NuGet.Protocol](https://www.nuget.org/packages/NuGet.Protocol) to work with authenticated feeds.
+
+```c#
+var sourceUri = "https://contoso.privatefeed/v3/index.json";
+var packageSource = new PackageSource(sourceUri)
+{
+    Credentials = new PackageSourceCredential(
+        source: sourceUri,
+        username: "myUsername",
+        passwordText: "myVerySecretPassword",
+        isPasswordClearText: true,
+        validAuthenticationTypesText: null)
+};
+
+// If the `SourceRepository` is created with a `PackageSource`, the rest of APIs will consume the credentials attached to `PackageSource.Credentials`.
+SourceRepository repository = Repository.Factory.GetCoreV3(packageSource);
+PackageMetadataResource resource = await repository.GetResourceAsync<PackageMetadataResource>();
+
+IEnumerable<IPackageSearchMetadata> packages = await resource.GetMetadataAsync(
+    "MyPackage",
+    includePrerelease: true,
+    includeUnlisted: false,
+    new SourceCacheContext(),
+    NullLogger.Instance,
+    CancellationToken.None);
+
+foreach (IPackageSearchMetadata package in packages)
+{
+    Console.WriteLine($"Version: {package.Identity.Version}");
+    Console.WriteLine($"Listed: {package.IsListed}");
+    Console.WriteLine($"Tags: {package.Tags}");
+    Console.WriteLine($"Description: {package.Description}");
+}
+```
+
+## Aditional documentation
+
+More information about the NuGet.Protocol library can be found on the [official Microsoft documentation page](https://learn.microsoft.com/nuget/reference/nuget-client-sdk#nugetprotocol).
```

Comment:
Consider also linking to the NuGet API docs: https://learn.microsoft.com/nuget/api/overview